e2e/instances/list/index.spec.ts

```diff
@@ -0,0 +1,388 @@
+    .getByRole("heading", { name: "Recently executed instances" })
+    .click(); // click on table header to close all tooltips opened
+  await createdReltime.hover({ force: true });
+  const createdAtTooltip = page.getByTestId(
+    `instance-row-absolute-created-time-${instance.instance}`
+  );
+  await expect(
+    createdAtTooltip,
+    "on hover, the absolute time should appear"
+  ).toBeVisible();
+
+  const updatedReltime = page.getByTestId(
+    `instance-row-relative-updated-time-${instance.instance}`
+  );
+  await expect(updatedReltime, "updateAd should be visible").toBeVisible();
+
+  await page
+    .getByRole("heading", { name: "Recently executed instances" })
+    .click(); // click on table header to close all tooltips opened
+  await updatedReltime.hover({ force: true });
+  const updatedAtTooltip = page.getByTestId(
+    `instance-row-absolute-updated-time-${instance.instance}`
+  );
+  await expect(
+    updatedAtTooltip,
+    "on hover, the absolute time should appear"
+  ).toBeVisible();
+
+  const workflowLink = page.getByTestId(
+    `instance-row-workflow-${instance.instance}`
+  );
+  await workflowLink.click();
+  await expect(
+    page,
+    "on click workflow, page should navigate to the workflow page"
+  ).toHaveURL(
+    `/${namespace}/explorer/workflow/active/${instanceDetail?.as.split(":")[0]}`
+  );
+
+  await page.goBack();
+  await instanceItemRow.click();
+  await expect(
+    page,
+    "on click row, page should navigate to the instance detail page"
+  ).toHaveURL(`/${namespace}/instances/${instance.instance}`);
+
+  //no pagination to be visibe
+  await expect(
+    page.getByTestId("instance-list-pagination"),
+    "there is no pagination when there aren't many instances"
+  ).not.toBeVisible();
+});
+
+test("test the pagination", async ({ page }) => {
+  test.setTimeout(200000);
+
+  await createWorkflow({
+    payload: childWorkflowContent,
+    urlParams: {
+      baseUrl: process.env.VITE_DEV_API_DOMAIN,
+      namespace,
+      name: "child.yaml",
+    },
+  });
+
+  await createWorkflow({
+    payload: parentWorkflowContent,
+    urlParams: {
+      baseUrl: process.env.VITE_DEV_API_DOMAIN,
+      namespace,
+      name: "parent.yaml",
+    },
+  });
+
+  const parentInstance = await runWorkflow({
+    urlParams: {
+      baseUrl: process.env.VITE_DEV_API_DOMAIN,
+      namespace,
+      path: "parent.yaml",
+    },
+  });
+
+  // there needs to be a discussion to check the total count
+  // expect(instancesList.instances.pageInfo.total, "length of the instance list should be 5").toBe(5);
+  await page.goto(`${namespace}/instances/`, { waitUntil: "networkidle" });
+
+  await expect(
+    page.getByTestId("pagination-wrapper"),
+    "there should be pagination component"
+  ).toBeVisible();
+
+  const btnPrev = page.getByTestId("pagination-btn-left");
+  const btnNext = page.getByTestId("pagination-btn-right");
+
+  //loop through all pages by clicking Next button
+  const testPages = [
```

Comment:
I think we should be very careful with randomness here. With this approach, a failing test is very hard to reproduce. If we have e.g. an issue on the last page, it will only be caught with a very low probability and only on certain test runs. And I actually have "sometimes" have problems with this test. 

![CleanShot 2023-07-20 at 16 40 13@2x](https://github.com/direktiv/direktiv-ui/assets/121789579/aa7111a1-1d23-4587-b480-f95e13177279)

Did you expierence any similar problems?

Maybe there was a misunderstanding in what we need to test. I think we should go through every page and check the state of the pagination on each one.

Reply:
I just spoke with Sebastian and discussed this problem with using random pages and that he suggested this approach this morning. I kind of missed that, sorry! 

We actually had a misunderstanding of what should be tested and now agreed on keeping it simple and make much less test on the pagination itself (we have unit tests in place for the general behavior of the pagination component). So please just generate instances for 2 and a half pages (it would be good to make the last page not completely filled with instances) and click through each of the pages and check the pagination and first (maybe also the last) item on the page.

I think this is a lot simpler to read as well, because we don't need to loop over all pages dynamically and rather just statically go to page 2 and 3.